### PR TITLE
[#6857]Platform: Preflight checks for S3 backup storage configuration

### PR DESCRIPTION
**Pull request by jitendra-12113**
Heading:
[#6857]Platform: Preflight checks for S3 backup storage configuration
Description:
We need to validate the S3 backup config before saving it.

Validate AWS credentials
Validate whether bucketname already exists or not
Note: If user selects IAM then we are not validating the config details.
Testing:

Create aws cloud provider
Go to config and the backup tab, select Amazon S3 config
Enter either wrong AWS aceess key/secret key or the random bucket
name, we will get appropriate error message
Added Unit Tests as well.

Previous REF PR: https://github.com/yugabyte/yugabyte-db/pull/8518 (We had to revert this due to itest failure)
Followup change Issue ID for update Config: https://github.com/yugabyte/yugabyte-db/issues/8824

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java [hunk 1]
```diff
@@ -90,6 +99,59 @@ public void doValidate(JsonNode data, ObjectNode errorsCollector) {
...
+        // Assuming bucket name will always start with s3:// otherwise that will be invalid
+        if (s3UriPath.length() < 5 || !s3UriPath.startsWith("s3://")) {
+          errorJson.set(fieldName, Json.newArray().add("Invalid s3UriPath format: " + s3UriPath));
+        } else {
+          try {
+            s3UriPath = s3UriPath.substring(5);
+            String[] bucketSplit = s3UriPath.split("/", 2);
+            String bucketName = null;
+            String prefix = null;
+            if (bucketSplit.length == 2) {
```
Comment (SergeyPotachev) [inline managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java:125]:
```
String bucketName = bucketSplit.length > 0 ? bucketSplit[0] : "";
String prefix = bucketSplit.length > 1 ? bucketSplit[1] : "";
```

**Review by SergeyPotachev — CHANGES_REQUESTED** [review]
One small fix is requested.

File: managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java (hunk 1 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java]: Done. Thanks!

**Comment by gaurav061** [comment]
Please find the build response for the UI.
`npm run build:ci

> YB-Customer@0.0.0 build:ci /home/hasher/Yugabyte/Dev/yugabyte-db/managed/ui
> CI=true GENERATE_SOURCEMAP=false react-scripts build

Creating an optimized production build...
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <3.10.0

YOUR TYPESCRIPT VERSION: 4.0.2

Please only submit bug reports when using the officially supported version.

=============
Compiled successfully.

File sizes after gzip:

  990.52 KB (+196 B)  build/static/js/2.b326d039.chunk.js
  216.75 KB (+9 B)    build/static/js/main.70bcfa02.chunk.js
  39.49 KB            build/static/css/2.26f75584.chunk.css
  30.12 KB            build/static/css/main.8690be0d.chunk.css
  737 B               build/static/js/runtime-main.cf7c1c94.js
`

File: managed/ui/src/components/config/Storage/StorageConfiguration.js [hunk 2]
```diff
@@ -200,6 +193,14 @@ class StorageConfiguration extends Component {
             // reset form after successful submission due to BACKUP_LOCATION value is shared across all tabs
             this.props.reset();
             this.props.fetchCustomerConfigs();
+
+            // Change to list view if form is successfully submitted.
```
Comment (nishantSharma459) [inline managed/ui/src/components/config/Storage/StorageConfiguration.js:197]: This change is in the `else` block that will handle the `create config`, but what about the `if` block handling the `update config`? I believe the same needs to be implemented there as well. Please check it out. And test it thoroughly.

Comment (gaurav061) [reply managed/ui/src/components/config/Storage/StorageConfiguration.js:197]: We will address this as part of follow-up PR. We will add a link for that before merging.

Comment (nishantSharma459) [reply managed/ui/src/components/config/Storage/StorageConfiguration.js:197]: Cool.

**Review by nishantSharma459 — APPROVED** [review]

**Pull request merged**